packages/dev/core/src/XR/webXRManagedOutputCanvas.ts

```diff
@@ -127,13 +127,15 @@ export class WebXRManagedOutputCanvas implements WebXRRenderTarget {
                         },
                         () => {
                             // fail silently
-                            reject("makeXRCompatible failed");
+                            Tools.Warn("Error executing makeXRCompatible. This does not mean that the session will work incorrectly.");
+                            resolve();
                         }
                     );
                 } else {
                     resolve();
                 }
             } catch (e) {
+                // if this fails - the exception will be caught and the promise will be rejected
                 reject(e);

```

Comment:
Just curious, why is this rejected if the only consumer just ignores any errors anyway? Seems like if this code used async/await and just caught and logged the exceptions and never returned a rejected promise it would be a little clearer and would simplify the code at the call site as well.

Reply:
This is a different case than the promise rejection, so I actually wanted the browser to show the error here. The reasoning being - if the promise failed it is a chrome issue with the makeXRCompatible function, which, under certain conditions, might fail. The errors caught in this line are unexpected errors. In this case, the browser will show a red error providing the rejected reason and then will continue when this promise is consumed.

Not sure if it the right decision though.